awx/main/models/workflow.py

```diff
@@ -318,8 +318,7 @@ def get_job_kwargs(self):
         for parent_node in self.get_parent_nodes():
             is_root_node = False
             aa_dict.update(parent_node.ancestor_artifacts)
-            if parent_node.job and hasattr(parent_node.job, 'artifacts'):
-                aa_dict.update(parent_node.job.artifacts)
+            aa_dict.update(parent_node.job.get_effective_artifacts(parents_set=set([self.workflow_job_id])))
```

Comment:
:eyes: 

I need to think through this.  This is the part that I worry is going to cause a bunch of repeated queries.

Reply:
Never mind, I think this isn't a problem.

I still dislike `set([...])`, though.